src/action/darwin/create_fstab_entry.rs

```diff
@@ -0,0 +1,237 @@
+            .await?
+            .ok_or_else(|| {
+                ActionError::Custom(Box::new(CreateFstabEntryError::NoVolume(
+                    apfs_volume_label.clone(),
+                )))
+            })?;
+        let fstab_entry = fstab_entry(&uuid, apfs_volume_label);
+
+        let mut file = OpenOptions::new()
+            .create(false)
+            .write(true)
+            .read(true)
+            .open(&fstab_path)
+            .await
+            .map_err(|e| ActionError::Open(fstab_path.to_owned(), e))?;
+
+        let mut file_contents = String::default();
+        file.read_to_string(&mut file_contents)
+            .await
+            .map_err(|e| ActionError::Read(fstab_path.to_owned(), e))?;
+
+        if let Some(start) = file_contents.rfind(fstab_entry.as_str()) {
+            let end = start + fstab_entry.len();
+            file_contents.replace_range(start..end, "")
+        }
+
+        file.seek(SeekFrom::Start(0))
+            .await
+            .map_err(|e| ActionError::Seek(fstab_path.to_owned(), e))?;
+        file.set_len(0)
+            .await
+            .map_err(|e| ActionError::Truncate(fstab_path.to_owned(), e))?;
+        file.write_all(file_contents.as_bytes())
+            .await
+            .map_err(|e| ActionError::Write(fstab_path.to_owned(), e))?;
+        file.flush()
+            .await
+            .map_err(|e| ActionError::Flush(fstab_path.to_owned(), e))?;
+
+        Ok(())
+    }
+}
+
+async fn get_uuid_for_label(apfs_volume_label: &str) -> Result<Option<Uuid>, ActionError> {
+    let output = execute_command(
+        Command::new("/usr/sbin/diskutil")
+            .process_group(0)
+            .arg("info")
+            .arg(apfs_volume_label)
+            .stdin(std::process::Stdio::null())
+            .stdout(std::process::Stdio::piped()),
+    )
+    .await
+    .map_err(|e| ActionError::Command(e))?;
+
+    let stdout = String::from_utf8(output.stdout)?;
+
+    let mut found = None;
+    for line in stdout.lines() {
+        let prefix = "Volume UUID:";
+        let trimmed = line.trim();
+        if let Some(index) = trimmed.find(prefix) {
+            let maybe_uuid = trimmed[(index + prefix.len())..].trim();
+            let uuid = Uuid::parse_str(maybe_uuid).map_err(|err| {
+                ActionError::Custom(Box::new(CreateFstabEntryError::Uuid(
+                    maybe_uuid.to_string(),
+                    err,
+                )))
+            })?;
+
+            found = Some(uuid);
+            break;
+        }
+    }
+
+    Ok(found)
+}
+
+fn fstab_prelude_comment(apfs_volume_label: &str) -> String {
+    format!("# nix-installer created volume labelled `{apfs_volume_label}`")
+}
+
+fn fstab_entry(uuid: &Uuid, apfs_volume_label: &str) -> String {
+    let prelude_comment = fstab_prelude_comment(apfs_volume_label);
+    format!(
+        "\
+        {prelude_comment}\n\
+        UUID=\"{uuid}\" /nix apfs rw,noauto,nobrowse,suid,owners\n\
```

Comment:
```suggestion
        UUID={uuid} /nix apfs rw,noauto,nobrowse,suid,owners\n\
```

Reported in the issue that the quotes cause issues.

Reply:
Good catch!